src/control/dmg/pool.go

```diff
@@ -23,7 +23,177 @@
 
 package main
 
-// TODO: provide implementation to this placeholder
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/daos-stack/daos/src/control/common"
+	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/inhies/go-bytesize"
+	"github.com/pkg/errors"
+)
+
+const (
+	msgSizeNoNumber = "size string doesn't specify a number"
+	msgSizeBad      = "size string illegal, %s"
+	msgSizeZeroScm  = "non-zero scm size is required"
+	maxNumSvcReps   = 13
+)
 
 // PoolCmd is the struct representing the top-level pool subcommand.
-type PoolCmd struct{}
+type PoolCmd struct {
+	Create CreatePoolCmd `command:"create" alias:"c" description:"Create a DAOS pool"`
+}
+
+// CreatePoolCmd is the struct representing the command to create a DAOS pool.
+type CreatePoolCmd struct {
+	GroupName  string `short:"g" long:"group" description:"DAOS pool to be owned by given group"`
```

Comment:
There is a name collision with the pool server process group --group= (not implemented in this patch but commented out below; yet needed I think), and this permissions/access related use of --group= . Can we consider something like one of the following to eliminate the name collision?:
--username= (alias "u")
--groupname= (alias "g")
--group= (alias "G" as noted in the commented-out line below)

or
--user= (alias "u")
--group= (alias "g")
--procgroup= (alias "G" or "p")

Reply:
I like the second but proc-group to be consistent and with alias G, thanks

Reply:
Johann and I discussed; the current --group=SRVGRPNAME is the name of the DAOS system. One possibility I can think of is to name the option --sys-name=. This would result in a broad search&replace of --group in both dmg and daos tools commands in the proposal.

Reply:
can we update this in a follow-on patch, when the iosrv side of this work lands?

Reply:
https://jira.hpdd.intel.com/browse/DAOS-2679